date_range/tests/test_date_range_search_mixin.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# Copyright 2021 Opener B.V. <stefan@opener.amsterdam>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from dateutil.rrule import MONTHLY
+
+from odoo.tests import SavepointCase
+from .common import setup_test_model, teardown_test_model
+from .models import TestDateRangeSearchMixin
+
+
+class TestDateRangeearchMixin(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        setup_test_model(cls.env, [TestDateRangeSearchMixin])
```

Comment:
@StefanRijnhart You could use [odoo-test-helper](https://pypi.org/project/odoo-test-helper/) to build your test model.

Reply:
Looks good! How do I deal with the additional library dependency? It would be awkward to add a new dependency to the module manifest, but if I only import it in tests such that it does not affect runtime code, will it be picked up by pre-commit hook for inclusion in requirements.txt?

Reply:
You just need to put the dependency into the `test-requirements.txt` file at the root of the repo. No need to add the dependency into the manifest. (ex https://github.com/OCA/search-engine/blob/14.0/test-requirements.txt)

Reply:
Perfect, will do!